modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/webhooks/credentials/CredentialsEventTestCase.java

```diff
@@ -0,0 +1,110 @@
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(userMode);
+
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+        identityGovernanceRestClient = new IdentityGovernanceRestClient(serverURL, tenantInfo);
+
+        UserObject userInfo = new UserObject()
+                .userName("test-user")
+                .password("TestPassword@123")
+                .name(new Name().givenName("test-user-given-name").familyName("test-user-last-name"))
+                .addEmail(new Email().value("test-user@test.com"));
+        userId = scim2RestClient.createUser(userInfo);
+
+        webhookEventTestManager = new WebhookEventTestManager("/scim2/webhook", "WSO2",
+                Arrays.asList("https://schemas.identity.wso2.org/events/user",
+                        "https://schemas.identity.wso2.org/events/credential"),
+                "CredentialsEventTestCase",
+                automationContext);
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        scim2RestClient.closeHttpClient();
+        identityGovernanceRestClient.closeHttpClient();
+        webhookEventTestManager.teardown();
+    }
+
+    @Factory(dataProvider = "testExecutionContextProvider")
+    public CredentialsEventTestCase(TestUserMode userMode) throws Exception {
+
+        this.userMode = userMode;
+        automationContext = new AutomationContext("IDENTITY", userMode);
+    }
+
+    @DataProvider(name = "testExecutionContextProvider")
+    public static Object[][] getTestExecutionContext() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_USER},
+                {TestUserMode.TENANT_USER}
+        };
+    }
+
+    @Test
+    public void testUserProfileUpdateWithoutAnyAccountStateManagementClaims() throws Exception {
+
+        assertNotNull(userId);
+
+        PatchOperationRequestObject patchRequest = new PatchOperationRequestObject();
+
+        UserItemAddGroupobj newPassword = new UserItemAddGroupobj().op(UserItemAddGroupobj.OpEnum.REPLACE);
+        newPassword.setPath("password");
+        newPassword.setValue("NewSecurePassword123!");
+        patchRequest.addOperations(newPassword);
+
+        scim2RestClient.updateUser(patchRequest, userId);
+
+        webhookEventTestManager.stackExpectedEventPayload(
+                "https://schemas.identity.wso2.org/events/credential/event-type/credentialUpdated",
+                AdminInitUserManagementEventTestExpectedEventPayloadBuilder.buildExpectedUserCredentialUpdatedEventPayloadWithoutAnyOtherClaims(
+                        userId, automationContext.getContextTenant().getDomain()));
+        webhookEventTestManager.validateStackedEventPayloads();
+    }
+
+    @Test(dependsOnMethods = {"testUserProfileUpdateWithoutAnyAccountStateManagementClaims"})
+    public void testDeleteUser() throws Exception {
+
+        scim2RestClient.deleteUser(userId);
```

Comment:
We can move this to Afterclass method, rather than having a separate test case

Reply:
Fixed